### PR TITLE
Case

### PR DESCRIPTION
**Pull request by sbreiff**
**In this PR:**
- added Case schema and types file (limited properties so far)
- added Case insert
- individual and sample_processing revlinks to Case
- tests for revlinks
- added bin/ files to .gitignore (same ones as fourfront)

**Update:**
- also added Report schema
- Report has Case revlink and display_title inherits from case id
- added some schema specific ids (case_id, family_id, individual_id)
- added tests for new calculated properties
- edited some inserts to have these identifiers

### CODE REVIEW

File: src/encoded/tests/datafixtures.py [hunk 1]
```diff
@@ -5,8 +5,8 @@
 ORDER = [
     'user', 'project', 'institution', 'file_format', 'variant_consequence', 'cohort', 'family', 'individual',
     'sample', 'workflow', 'access_key', 'disorder', 'document', 'file_fastq',
-    'file_processed', 'file_reference', 'gene', 'sample_processing',
-    'page', 'phenotype', 'quality_metric_fastqc', 'evidence_dis_pheno',
+    'file_processed', 'file_reference', 'gene', 'sample_processing', 'case',
+    'report', 'page', 'phenotype', 'quality_metric_fastqc', 'evidence_dis_pheno',
```
Comment (aschroed) [inline src/encoded/tests/datafixtures.py:9]: Thinking you might want to move phenotype earlier in the ORDER but probably doesn't matter all that much

File: src/encoded/types/__init__.py [hunk 2]
```diff
@@ -39,6 +41,55 @@ class SampleProcessing(Item):
...
+    @calculated_property(schema={
+        "title": "Case",
+        "description": "The case this sample processing is for",
+        "type": "string",
+        "linkTo": "Case"
+    })
+    def case(self, request):
+        rs = self.rev_link_atids(request, "case")
+        if rs:
+            return rs[0]
```
Comment (aschroed) [inline src/encoded/types/__init__.py:55]: There can be more than one case arranged so the first one is always the correct one?

File: src/encoded/types/__init__.py [hunk 3]
```diff
@@ -39,6 +41,55 @@ class SampleProcessing(Item):
...
```
Comment (aschroed) [inline src/encoded/types/__init__.py:79]: same question as above.

File: src/encoded/schemas/sample_processing.json [hunk 4]
```diff
@@ -21,6 +21,18 @@
         "schema_version": {
             "default": "1"
         },
+        "analysis_type": {
```
Comment (KorayKirli) [inline src/encoded/schemas/sample_processing.json:24]: How would we phrase analysis done with
mother-proband
father-proband
mother-father-proband-brother
....

While trio refers to mother-father-proband there are other options too

Is there a use in being more generic, or having an additional generic option (WGS-Group), or having this calculated. WGS WES part can come from the sample.

Comment (sbreiff) [reply src/encoded/schemas/sample_processing.json:24]: Those are good points, maybe for now i'll make it a suggested enum adding WGS-Group, and once i figure out what we would want displayed in the Case table for the different options i can edit this or turn it into a calculated property, if possible.

**Pull request merged**